### PR TITLE
Add NusaParagraph Topic Classification

### PR DESCRIPTION
**Pull request by gentaiscool**
NusaParagraphTopicClassification is a multi-class topic classification on 10 Indonesian languages.

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: ... 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

cc @Muennighoff @KennethEnevoldsen

### CODE REVIEW

File: mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            month = nov,
+            year = "2023",
+            address = "Nusa Dua, Bali",
+            publisher = "Association for Computational Linguistics",
+            url = "https://aclanthology.org/2023.ijcnlp-main.60",
+            pages = "921--945",
+        }
+        """,
+        n_samples={"train": 15516, "validation": 2948, "test": 6250},
+        avg_character_length={"train": 5.51, "validation": 5.49, "test": 5.45},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py:59]: again sizes seems quite small

File: mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        name="NusaParagraphTopicClassification",
+        dataset={
+            "path": "gentaiscool/nusaparagraph_topic",
+            "revision": "abb43f8d5b9510b8724b48283aca26c4733eac5d",
+        },
+        description="NusaParagraphTopicClassification is a multi-class topic classification on 10 Indonesian languages.",
+        reference="https://github.com/IndoNLP/nusa-writes",
+        category="s2s",
+        type="Classification",
+        eval_splits=["train", "validation", "test"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py:32]:
```suggestion
        eval_splits=["test"],
```
seems like we can test with just the test set.

File: mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py (hunk 1 above)
Comment (gentaiscool) [reply mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py:59]: let me update the sizes (previously I put the avg number of char per word)

**Comment by gentaiscool** [comment]
Thank you for the review. I addressed them. I updated the avg number of character per sample, added the results, and points. I calculated the points, so there are 4 new languages `bew-Latn`, `mak-Latn`, `mui-Latn`, `rej-Latn` (2 + 4 x 4 = 18). Plus, I added the point for the reviewer. @KennethEnevoldsen

**Pull request merged**